### DISCUSSION TITLE
[PSD16] Esercizio Slide 18

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#219] [Esercizi (A-L)]
![image](https://user-images.githubusercontent.com/69389461/215794165-fc27c07e-e7e6-4922-a89f-f8f126ec3bf7.png)

**Comment by Elia-Belli** [comment]
### Frequenza Massima
La frequenza è l'inverso del periodo (del clock), la disciplina temporale ci impone alcuni vincoli sul periodo minimo del clock $T_c$ affinchè il circuito operi in modo affidabile, ed applicando i vari vincoli si arriva alla disequazione: 
$$T_c \geq t_{pcq} + t_{pd} + t_{su}$$
Ogni volta che si calcola un delay si valuta sempre il caso peggiore, ovvero il **critical path** (attenzione, va considerato il percorso più lungo compreso tra 2 fliflop), quindi in questo caso $t_{pd} = 2 \times t_{gate\ pd}$ (anche se le porte sono diverse l'esercizio assegna a tutte lo stesso delay), di conseguenza:
$T_c \geq 160ps \rightarrow F_{max} = 1/160\ ps^{-1} = 6.25\ GHz$
### Latenza e Portata
La latenza $L$ è il tempo impiegato ad elaborare un token (un input o insieme di input), ovvero il tempo che passa dall'entrata del token nel circuito alla sua uscita, in questo caso la latenza è $L = 160\ ps$ (dal secondo token in poi), di conseguenza la portata è di $6.25 \times 10^9\ token/s$.

Rispetto all'esercizio della Slide 17 (#218 ), abbiamo fatto uso del parallelismo temporale per ridurre la latenza ed aumentare la portata.
